`change_point` agg fails with `IllegalArgumentException` while indexing an array

### Elasticsearch Version

8.16.0-SNAPSHOT

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

Linux 6.5.0-1024-gcp #26~22.04.1-Ubuntu SMP Fri Jun 14 18:48:45 UTC 2024 x86_64 GNU/Linux

### Problem Description

When run on some sets of documents the `change_point` aggregation throws an `IllegalArgumentException`. The Observability Logs UX team is trying to use the aggregation to detect change points in log documents. I was unable to detect a pattern to the failures, though, as slight modifications to the buckets cause it to disappear or re-appear:

```json
{
  "error": {
    "root_cause": [],
    "type": "search_phase_execution_exception",
    "reason": "",
    "phase": "rank-feature",
    "grouped": true,
    "failed_shards": [],
    "caused_by": {
      "type": "illegal_argument_exception",
      "reason": "0 > -1",
      "stack_trace": """java.lang.IllegalArgumentException: 0 > -1
	at java.base/java.util.Arrays.copyOfRange(Arrays.java:4090)
	at org.elasticsearch.ml@8.16.0-SNAPSHOT/org.elasticsearch.xpack.ml.aggs.changepoint.ChangePointAggregator.testDistributionChange(ChangePointAggregator.java:487)
	at org.elasticsearch.ml@8.16.0-SNAPSHOT/org.elasticsearch.xpack.ml.aggs.changepoint.ChangePointAggregator.testForChange(ChangePointAggregator.java:241)
	at org.elasticsearch.ml@8.16.0-SNAPSHOT/org.elasticsearch.xpack.ml.aggs.changepoint.ChangePointAggregator.testForChange(ChangePointAggregator.java:160)
	at org.elasticsearch.ml@8.16.0-SNAPSHOT/org.elasticsearch.xpack.ml.aggs.changepoint.ChangePointAggregator.doReduce(ChangePointAggregator.java:129)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.search.aggregations.InternalAggregations.topLevelReduce(InternalAggregations.java:235)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.search.aggregations.InternalAggregations.topLevelReduceDelayable(InternalAggregations.java:211)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.action.search.SearchPhaseController.reduceAggs(SearchPhaseController.java:682)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.action.search.SearchPhaseController.reducedQueryPhase(SearchPhaseController.java:636)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.action.search.QueryPhaseResultConsumer.reduce(QueryPhaseResultConsumer.java:139)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.action.search.RankFeaturePhase.innerRun(RankFeaturePhase.java:92)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.action.search.RankFeaturePhase$1.doRun(RankFeaturePhase.java:79)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:991)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
"""
    },
    "stack_trace": """Failed to execute phase [rank-feature], 
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseFailure(AbstractSearchAsyncAction.java:726)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.action.search.RankFeaturePhase$1.onFailure(RankFeaturePhase.java:84)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:28)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:991)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
Caused by: java.lang.IllegalArgumentException: 0 > -1
	at java.base/java.util.Arrays.copyOfRange(Arrays.java:4090)
	at org.elasticsearch.ml@8.16.0-SNAPSHOT/org.elasticsearch.xpack.ml.aggs.changepoint.ChangePointAggregator.testDistributionChange(ChangePointAggregator.java:487)
	at org.elasticsearch.ml@8.16.0-SNAPSHOT/org.elasticsearch.xpack.ml.aggs.changepoint.ChangePointAggregator.testForChange(ChangePointAggregator.java:241)
	at org.elasticsearch.ml@8.16.0-SNAPSHOT/org.elasticsearch.xpack.ml.aggs.changepoint.ChangePointAggregator.testForChange(ChangePointAggregator.java:160)
	at org.elasticsearch.ml@8.16.0-SNAPSHOT/org.elasticsearch.xpack.ml.aggs.changepoint.ChangePointAggregator.doReduce(ChangePointAggregator.java:129)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.search.aggregations.InternalAggregations.topLevelReduce(InternalAggregations.java:235)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.search.aggregations.InternalAggregations.topLevelReduceDelayable(InternalAggregations.java:211)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.action.search.SearchPhaseController.reduceAggs(SearchPhaseController.java:682)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.action.search.SearchPhaseController.reducedQueryPhase(SearchPhaseController.java:636)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.action.search.QueryPhaseResultConsumer.reduce(QueryPhaseResultConsumer.java:139)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.action.search.RankFeaturePhase.innerRun(RankFeaturePhase.java:92)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.action.search.RankFeaturePhase$1.doRun(RankFeaturePhase.java:79)
	at org.elasticsearch.server@8.16.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	... 6 more
"""
  },
  "status": 400
}
```

### Steps to Reproduce

I originally encountered this when running on millions of log entries, but I could reduce it to this synthetic scenario:

```
DELETE change-point-test

POST _bulk
{ "index" : { "_index" : "change-point-test", "_id" : "1" } }
{ "key" : 1, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "2" } }
{ "key" : 2, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "3" } }
{ "key" : 3, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "4" } }
{ "key" : 4, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "5" } }
{ "key" : 5, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "6" } }
{ "key" : 6, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "7" } }
{ "key" : 7, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "8" } }
{ "key" : 8, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "9" } }
{ "key" : 9, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "10" } }
{ "key" : 10, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "11" } }
{ "key" : 11, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "12" } }
{ "key" : 12, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "13" } }
{ "key" : 13, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "14" } }
{ "key" : 14, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "15" } }
{ "key" : 15, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "16" } }
{ "key" : 16, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "17" } }
{ "key" : 17, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "18" } }
{ "key" : 18, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "19" } }
{ "key" : 19, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "20" } }
{ "key" : 20, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "21" } }
{ "key" : 21, "value" : 0 }
{ "index" : { "_index" : "change-point-test", "_id" : "22" } }
{ "key" : 22, "value" : 700 }
{ "index" : { "_index" : "change-point-test", "_id" : "23" } }
{ "key" : 23, "value" : 735 }
{ "index" : { "_index" : "change-point-test", "_id" : "24" } }
{ "key" : 24, "value" : 715 }
{ "index" : { "_index" : "change-point-test", "_id" : "25" } }
{ "key" : 25, "value" : 0 }

POST change-point-test/_search?error_trace
{
  "size": 0,
  "aggs": {
    "buckets": {
      "terms": {
        "field": "key",
        "size": 100
      },
      "aggs": {
        "values": {
          "max": {
            "field": "value",
            "missing": 0
          }
        }
      }
    },
    "change": {
      "change_point": {
        "buckets_path": "buckets>values"
      }
    }
  }
}
```

### Logs (if relevant)

_No response_

We're encountering a very similar error in 8.14.3 - I will see if I can get a reproduction case later, though it's on an internal cluster so ML team can get access if required 
Request:
```
GET rally-results-*/_search?error_trace
{
  "query": {
    "bool": {
      "filter": [
        {
          "query_string": {
            "query": "* AND environment:\"serverless-nightly\" AND active:true AND user-tags.benchmark-name:\"serverless-qa-aws-so_vector\" AND merge_throttle_time"
          }
        },
        {
          "range": {
            "race-timestamp": {
              "gte": "2024-11-18T10:08:35+00:00||-30d/d",
              "lte": "2024-11-18T10:08:35+00:00",
              "format": "strict_date_optional_time"
            }
          }
        }
      ]
    }
  },
  "size": 0,
  "aggs": {
    "race_timestamp_term": {
      "terms": {
        "field": "race-timestamp",
        "size": 30,
        "order": {
          "_key": "asc"
        }
      },
      "aggs": {
        "metric": {
          "avg": {
            "field": "value.single"
          }
        }
      }
    },
    "changes": {
      "change_point": {
        "buckets_path": "race_timestamp_term>metric.value"
      }
    }
  }
}
```
Error 
```
{
  "error": {
    "root_cause": [],
    "type": "search_phase_execution_exception",
    "reason": "",
    "phase": "fetch",
    "grouped": true,
    "failed_shards": [],
    "caused_by": {
      "type": "illegal_argument_exception",
      "reason": "0 > -1",
      "stack_trace": """java.lang.IllegalArgumentException: 0 > -1
	at java.base/java.util.Arrays.copyOfRange(Arrays.java:4090)
	at org.elasticsearch.ml@8.14.3/org.elasticsearch.xpack.ml.aggs.changepoint.ChangePointAggregator.testDistributionChange(ChangePointAggregator.java:487)
	at org.elasticsearch.ml@8.14.3/org.elasticsearch.xpack.ml.aggs.changepoint.ChangePointAggregator.testForChange(ChangePointAggregator.java:241)
	at org.elasticsearch.ml@8.14.3/org.elasticsearch.xpack.ml.aggs.changepoint.ChangePointAggregator.testForChange(ChangePointAggregator.java:160)
	at org.elasticsearch.ml@8.14.3/org.elasticsearch.xpack.ml.aggs.changepoint.ChangePointAggregator.doReduce(ChangePointAggregator.java:129)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.search.aggregations.InternalAggregations.topLevelReduce(InternalAggregations.java:246)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.search.aggregations.InternalAggregations.topLevelReduceDelayable(InternalAggregations.java:222)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.action.search.SearchPhaseController.reduceAggs(SearchPhaseController.java:678)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.action.search.SearchPhaseController.reducedQueryPhase(SearchPhaseController.java:632)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.action.search.QueryPhaseResultConsumer.reduce(QueryPhaseResultConsumer.java:139)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.action.search.FetchSearchPhase.innerRun(FetchSearchPhase.java:99)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.action.search.FetchSearchPhase$1.doRun(FetchSearchPhase.java:87)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
"""
    },
    "stack_trace": """Failed to execute phase [fetch], 
	at org.elasticsearch.server@8.14.3/org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseFailure(AbstractSearchAsyncAction.java:712)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.action.search.FetchSearchPhase$1.onFailure(FetchSearchPhase.java:92)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:28)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
Caused by: java.lang.IllegalArgumentException: 0 > -1
	at java.base/java.util.Arrays.copyOfRange(Arrays.java:4090)
	at org.elasticsearch.ml@8.14.3/org.elasticsearch.xpack.ml.aggs.changepoint.ChangePointAggregator.testDistributionChange(ChangePointAggregator.java:487)
	at org.elasticsearch.ml@8.14.3/org.elasticsearch.xpack.ml.aggs.changepoint.ChangePointAggregator.testForChange(ChangePointAggregator.java:241)
	at org.elasticsearch.ml@8.14.3/org.elasticsearch.xpack.ml.aggs.changepoint.ChangePointAggregator.testForChange(ChangePointAggregator.java:160)
	at org.elasticsearch.ml@8.14.3/org.elasticsearch.xpack.ml.aggs.changepoint.ChangePointAggregator.doReduce(ChangePointAggregator.java:129)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.search.aggregations.InternalAggregations.topLevelReduce(InternalAggregations.java:246)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.search.aggregations.InternalAggregations.topLevelReduceDelayable(InternalAggregations.java:222)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.action.search.SearchPhaseController.reduceAggs(SearchPhaseController.java:678)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.action.search.SearchPhaseController.reducedQueryPhase(SearchPhaseController.java:632)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.action.search.QueryPhaseResultConsumer.reduce(QueryPhaseResultConsumer.java:139)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.action.search.FetchSearchPhase.innerRun(FetchSearchPhase.java:99)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.action.search.FetchSearchPhase$1.doRun(FetchSearchPhase.java:87)
	at org.elasticsearch.server@8.14.3/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	... 6 more
"""
  },
  "status": 400
} 
```

--- a/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/changepoint/ChangeDetector.java
+++ b/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/changepoint/ChangeDetector.java
@@ -378,10 +378,12 @@ public class ChangeDetector {
         // before we run the tests.
         SampleData sampleData = sample(values, weights, discoveredChangePoints);
         final double[] sampleValues = sampleData.values();
-        final double[] sampleWeights = sampleData.weights();
 
         double pValue = 1;
         for (int cp : sampleData.changePoints()) {
+            if (cp == -1) {
+                continue;
+            }
             double[] x = Arrays.copyOfRange(sampleValues, 0, cp);
             double[] y = Arrays.copyOfRange(sampleValues, cp, sampleValues.length);
             double statistic = KOLMOGOROV_SMIRNOV_TEST.kolmogorovSmirnovStatistic(x, y);

--- a/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/changepoint/ChangeDetector.java
+++ b/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/changepoint/ChangeDetector.java
@@ -189,7 +189,7 @@ public class ChangeDetector {
     private TestStats testStepChangeVs(TestStats H0, double[] values, double[] weights, int[] candidateChangePoints) {
 
         double vStep = Double.MAX_VALUE;
-        int changePoint = -1;
+        int changePoint = ChangeType.NO_CHANGE_POINT;
 
         // Initialize running stats so that they are only missing the individual changepoint values
         RunningStats lowerRange = new RunningStats();
@@ -226,7 +226,7 @@ public class ChangeDetector {
     private TestStats testTrendChangeVs(TestStats H0, double[] values, double[] weights, int[] candidateChangePoints) {
 
         double vChange = Double.MAX_VALUE;
-        int changePoint = -1;
+        int changePoint = ChangeType.NO_CHANGE_POINT;
 
         // Initialize running stats so that they are only missing the individual changepoint values
         RunningStats lowerRange = new RunningStats();
@@ -349,7 +349,7 @@ public class ChangeDetector {
     ) {
 
         double maxDiff = 0.0;
-        int changePoint = -1;
+        int changePoint = ChangeType.NO_CHANGE_POINT;
 
         // Initialize running stats so that they are only missing the individual changepoint values
         RunningStats lowerRange = new RunningStats();
@@ -381,7 +381,7 @@ public class ChangeDetector {
 
         double pValue = 1;
         for (int cp : sampleData.changePoints()) {
-            if (cp == -1) {
+            if (cp == ChangeType.NO_CHANGE_POINT) {
                 continue;
             }
             double[] x = Arrays.copyOfRange(sampleValues, 0, cp);
@@ -453,7 +453,7 @@ public class ChangeDetector {
         }
 
         TestStats(Type type, double pValue, double var, double nParams, DataStats dataStats) {
-            this(type, pValue, var, nParams, -1, dataStats);
+            this(type, pValue, var, nParams, ChangeType.NO_CHANGE_POINT, dataStats);
         }
 
         boolean accept(double pValueThreshold) {

--- a/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/changepoint/ChangePointAggregator.java
+++ b/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/changepoint/ChangePointAggregator.java
@@ -47,7 +47,7 @@ public class ChangePointAggregator extends SiblingPipelineAggregator {
         ChangeType change = ChangePointDetector.getChangeType(bucketValues);
 
         ChangePointBucket changePointBucket = null;
-        if (change.changePoint() >= 0) {
+        if (change.changePoint() != ChangeType.NO_CHANGE_POINT) {
             changePointBucket = extractBucket(bucketsPaths()[0], aggregations, change.changePoint()).map(
                 b -> new ChangePointBucket(b.getKey(), b.getDocCount(), b.getAggregations())
             ).orElse(null);

--- a/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/changepoint/ChangeType.java
+++ b/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/changepoint/ChangeType.java
@@ -21,8 +21,10 @@ import java.util.Objects;
  */
 public interface ChangeType extends NamedWriteable, NamedXContentObject {
 
+    int NO_CHANGE_POINT = -1;
+
     default int changePoint() {
-        return -1;
+        return NO_CHANGE_POINT;
     }
 
     default double pValue() {

--- a/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/aggs/changepoint/ChangeDetectorTests.java
+++ b/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/aggs/changepoint/ChangeDetectorTests.java
@@ -17,6 +17,7 @@ import org.elasticsearch.xpack.ml.aggs.MlAggsHelper;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.stream.DoubleStream;
 
+import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThan;
 import static org.hamcrest.Matchers.instanceOf;
 import static org.hamcrest.Matchers.lessThan;
@@ -242,5 +243,15 @@ public class ChangeDetectorTests extends AggregatorTestCase {
         );
         ChangeType type = new ChangeDetector(bucketValues).detect(0.05);
         assertThat(type, instanceOf(ChangeType.DistributionChange.class));
+    }
+
+    public void testUncertainNonStationary() {
+        MlAggsHelper.DoubleBucketValues bucketValues = new MlAggsHelper.DoubleBucketValues(
+            null,
+            new double[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 700, 735, 715 }
+        );
+        ChangeType type = new ChangeDetector(bucketValues).detect(0.01);
+        assertThat(type, instanceOf(ChangeType.NonStationary.class));
+        assertThat(((ChangeType.NonStationary) type).getTrend(), equalTo("increasing"));
     }
 }

